lib/rules/generator-star-spacing.js

```diff
@@ -126,17 +214,23 @@ module.exports = {
             }
 
             const starToken = getStarToken(node);
-
-            // Only check before when preceded by `function`|`static` keyword
             const prevToken = sourceCode.getTokenBefore(starToken);
+            const nextToken = sourceCode.getTokenAfter(starToken);
 
-            if (prevToken.value === "function" || prevToken.value === "static") {
-                checkSpacing("before", prevToken, starToken);
+            let kind = "named";
+
+            if (node.type === "FunctionExpression" && (node.parent.type === "MethodDefinition" || node.parent.type === "Property")) {
+                kind = prevToken.value === "static" ? "static" : "method";
```

Comment:
I don't think this check for methods is correct. It's possible for objects to have function properties that aren't methods:

```js
var foo = {
  bar: function* () {
  }
};
```

That should fall under `anonymous`, not `method`.

Instead, I think it would be better to check the `.method` property of the parent node when `node.parent.type === "Property"`. The `method` property will be `true` if the node is a method, and `false` otherwise.